### PR TITLE
Moved http settings to the new settings infrastructure

### PR DESCRIPTION
**Pull request by colings86**
Moved http settings to the new settings infrastructure

```
The following settings were moved to Setting contents in HttpTransportSettings:

* http.cors.allow-origin
* http.port
* http.publish_port
* http.detailed_errors.enabled
* http.max_content_length
* http.max_chunk_size
* http.max_header_size
* http.max_initial_line_length

The following settings were removed:

* http.port
* http.netty.port
* http.netty.publish_port
* http.netty.max_content_length
* http.netty.max_chunk_size
* http.netty.max_header_size
* http.netty.max_initial_line_length
```

### CODE REVIEW

**Comment by s1monw** [comment]
@rmuir the static initialization commetn you made on the other PR from boaz applies here too?

**Comment by rmuir** [comment]
Yeah. We really need to factor netty to a module...

**Comment by s1monw** [comment]
@bleskes @colings86 can you move all these settings into a separate class such that we don't load any netty code on static init?

**Comment by colings86** [comment]
@s1monw I pushed a commit that moves the HTTP settings constants into a dedicated class

File: core/src/main/java/org/elasticsearch/http/HttpTransportSettings.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Setting.Scope;
+import org.elasticsearch.common.transport.PortsRange;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+
+public final class HttpTransportSettings {
+
+    public static final Setting<Boolean> SETTING_CORS_ENABLED = Setting.boolSetting("http.cors.enabled", false, false, Scope.CLUSTER);
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/http/HttpTransportSettings.java]: I think we can drop all the `*.netty.*` if we have an alternative without netty?

**Comment by s1monw** [comment]
left one comment - LGTM

**Comment by s1monw** [comment]
LGTM

**Pull request merged**